yt/visualization/accumulators.py

```diff
@@ -0,0 +1,348 @@
+
+    ..math::
+        I = \lim_{N \rightarrow \infty}\sum_{p=1}^N\sum_{i=1}^n a_{p,i}(r_{p+1,i}-r_{p,i})
+        
+    where :math:`n` is the number of dimensions.
+
+    This can be done as a matrix operation. If :math:`\vec{r} \equiv \vec{r}_{p+1} - \vec{r}_p`
+    then the above sum is (leaving out the limit):
+    
+    ..math::
+    
+        I = \vec{a}_1 \dot \vec{r}_1 + \vec{a}_2 \cdot \vec{r}_2 \ldots \vec{a}_N \cdot \vec{r}_N
+        
+    If we write the matrix :math:`A = ([a1x, a1y, ...], [a2x, a2y, ...], ...)`
+    and :math:`R = ([r1x, r1y, ...], [r2x, r2y, ...], ...)^T`, then the dot
+    products in the sum are the diagonal elements of the resulting matrix
+    multiplication :math:`AR`. The accumulation is then obtained by doing a
+    cumsum of this diagonal.
+
+    Parameters
+    ----------
+    p : YTArray 
+        The path to be integrated along
+
+    field_vals : YTArray 
+        An array containing the components of the vector field to be
+        integrated. The values are sampled at the starting point of
+        each path segment as well as the endpoint for the last segment
+
+    Returns
+    -------
+        accum : YTArray 
+            The cumulative value of the field integral at each path
+            segment
+    """
+    # https://en.wikipedia.org/wiki/Line_integral
+    # np.dot doesn't combine units correctly and there does not appear to be a
+    # unyt implementation of dot, so units will be handled manually
+    # for the time being
+    accum = np.cumsum(np.diag(np.dot(field_vals[:-1].d, (path[1:].d - path[:-1].d).T)))
+    accum = YTArray(accum, field_vals.units * path.units)
+    return accum
+
+
+def _accumulate_scalar_field(p, field_vals):
+    r"""
+    This function integrates a scalar field along a path. It uses a
+    similar method to that in _accumulate_vector_field, but the integral
+    is now:
+    
+    ..math::
+    
+        I = \int_C \phi(x1,x2,...,xn)d\vec{r}
+
+    Parameters
+    ----------
+    p : YTArray 
+        The path to be integrated along
+
+    fieldVals : YTArray 
+        An array containing the values of the scalar field to be
+        integrated at the location of the starting point of each
+        path segment as well as the endpoint for the last segment
+
+    Returns
+    -------
+    accum : YTArray 
+        The cumulative value of the field integral at each path
+        segment 
+    """
+    # https://en.wikipedia.org/wiki/Line_integral
+    # np.linalg.norm returns a ndarray, so when multiplied by field_vals, which
+    # is a YTArray, this leads to incorrect units. There does not appear to be
+    # a unyt implementation of norm, as far as I'm aware, so units will be
+    # handled manually for the time being
+    accum = np.cumsum(field_vals[:-1].d * np.linalg.norm(p[1:].d - p[:-1].d, axis=1))
+    accum = YTArray(accum, field_vals.units * p.units)
+    return accum
+
+
+def get_row_major_index(ncells, cell_ind):
```

Comment:
Is this the same functionality as `np.unravel_index` and `np.ravel_multi_index`?  If so, we might be able to ditch it.

Reply:
Not sure, as I've never heard of those numpy functions before haha! I'll take a look, and if they are the same then I'll use those! Thanks!

Reply:
It's the same as `np.ravel_multi_index` so I'll use that instead.